oxenstored/connections.ml

```diff
@@ -94,11 +119,35 @@ let del_watches cons con =
     |> Connection.Watch.Set.filter @@ fun w -> Connection.get_con w != con
     )
 
-let del_anonymous cons con =
+let del_anonymous cons con spec_fds =
   try
     Hashtbl.remove cons.anonymous (Connection.get_fd con) ;
-    del_watches cons con ;
-    Connection.close con
+    (* Reallocate the poll_status array, update indices pointing to it *)
+    cons.poll_status <-
```

Comment:
If you know the index that you are deleting this could also be done with 2 array copies. But I think we open/close these anon connections fairly infrequently, so this isn't in a critical path.

Reply:
Yes, that was my expectation as well.